packages/monaco-editor-wrapper/src/editorAppBase.ts

```diff
@@ -14,24 +11,29 @@ export type ModelUpdate = {
     codeOriginalUri?: string;
 }
 
-export type EditorAppBaseConfig = ModelUpdate & {
+export type EditorAppType = 'extended' | 'classic';
+
+export type EditorAppConfigBase = ModelUpdate & {
+    $type: EditorAppType;
     useDiffEditor: boolean;
     domReadOnly?: boolean;
     readOnly?: boolean;
-    awaitExtensionReadiness?: Array<() => Promise<void>>
-    userConfiguration?: UserConfiguration;
+    awaitExtensionReadiness?: Array<() => Promise<void>>;
+    overrideAutomaticLayout?: boolean;
+    editorOptions?: editor.IStandaloneEditorConstructionOptions;
+    diffEditorOptions?: editor.IStandaloneDiffEditorConstructionOptions;
 }
 
-export type EditorAppType = 'vscodeApi' | 'classic';
-
-export type UserConfiguration = {
-    json?: string;
+export enum ModelUpdateType {
+    none,
+    code,
+    model
```

Comment:
I think for TS it's a good idea to keep the enum values uppercased, although not required, I can see it was like this before as well.

Reply:
Yes, you are right.

Reply:
Done ⬇️